obt/link.c

```diff
@@ -170,15 +170,14 @@ ObtLink* obt_link_from_ddfile(const gchar *ddname, GSList *paths,
 
         if ((v = g_hash_table_lookup(keys, "Categories"))) {
             gulong i;
-            gchar *end;
 
             link->d.app.categories = g_new(GQuark, v->value.strings.n);
             link->d.app.n_categories = v->value.strings.n;
 
             for (i = 0; i < v->value.strings.n; ++i) {
                 link->d.app.categories[i] =
                     g_quark_from_string(v->value.strings.a[i]);
-                c = end = end+1; /* next */
+                c++; /* next */
```

Comment:
This looks really weird, but was also weird before, using an uninitialized |end| so I would not make this change on the suggestion of the tool alone, because now it actually looks intentional as this but I'm not really sure that is what it intended to set |c| to here. Could you add a TODO comment about this being wrong instead of making this change? Or can you explain why |c++| is the correct thing here..sorry the code is not well commented so it's a headache to follow now.

Reply:
Yes, inspecting it again it seems 'c++' is not the right thing to do.
The previous behaviour (until commit b025a0268f302df78589d3cc36aad5f56693799f#diff-aab854e268b36f6ae9d58283ef69534c) was to point to the end of the string, not the next char.
c is no longer used in the function, so actually this line should just be removed.
I'll update the PR tonight or tomorrow.